Cargo.toml

```diff
@@ -48,6 +48,7 @@ ntapi  = "0.3"
 [dev-dependencies]
 env_logger = { version = "0.6.2", default-features = false }
 rand = "0.4"
+socket2 = "0.3.15"
```

Comment:
This is a bad idea, socket2 is **not** ready for use.

Reply:
btw, why do you say it is not ready for use?

Reply:
Commits such as https://github.com/alexcrichton/socket2-rs/commit/7bf31f95af1905880e73a169b0eb3894597e6f0f, https://github.com/alexcrichton/socket2-rs/commit/b3badc7d3d599746a3632666876f1a253476c3c7 are only 1/2 releases old. I'm working on a proper review/rewrite, but I just don't have enough time.